Assets/Scripts/Algorithms/Patrolling/CyclicAlgorithms/SingleCycleChristofides.cs

```diff
@@ -0,0 +1,248 @@
+// Copyright 2025 MAEPS
+// 
+// This file is part of MAEPS
+// 
+// MAEPS is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+// 
+// MAEPS is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Henrik van Peet,
+// Mads Beyer Mogensen,
+// Puvikaran Santhirasegaram
+
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Map;
+
+namespace Maes.Algorithms.Patrolling
+{
+    /// <summary>
+    /// Implementation of the Single Cycle algorithm but using the solver approximation christofides as in: https://doi.org/10.1007/978-3-540-28645-5_48.
+    /// Christofides runs in O(n^3 lg n) time, where n is the number of vertices in the partition.
+    /// For reference (on my machine):
+    /// For 21 vertices, it took 8 seconds to compute the cycle.
+    /// For 9 vertices, it took 0.2 seconds to compute the cycle.
+    /// An implementation can be found here: https://github.com/matteoprata/DRONET-for-Patrolling/blob/main_july_2023/src/patrolling/tsp_cycle.py
+    /// </summary>
+    public sealed class SingleCycleChristofides : BaseCyclicAlgorithm
+    {
+        public override string AlgorithmName => "SingleCycleChristofides Algorithm";
+
+        /// <summary>
+        /// Use Christofides to get an approximate solution to TSP to cycle of all vertices in this robots partition.
+        /// </summary>
+        /// <param name="startVertex"></param>
+        protected override List<Vertex> CreatePatrollingCycle(Vertex startVertex)
+        {
+            var verticesInPartition = _patrollingMap.Vertices.Where(v => v.Partition == startVertex.Partition).ToList();
+            var estimatedDistanceMatrix = EstimatedDistanceMatrix(verticesInPartition);
+            var solution = Christofides(verticesInPartition, estimatedDistanceMatrix);
+            return solution.Select(id => verticesInPartition.Single(v => v.Id == id)).ToList();
+        }
+
+        public class Edge
```

Comment:
Are you sure you want this to be public? You should also consider making it an immutable struct.

Reply:
Please also add a constructor

Reply:
Done